microcosm_postgres/context.py

```diff
@@ -10,30 +10,38 @@
 from microcosm_postgres.operations import new_session, recreate_all
 
 
-class SessionContext:
-    """
-    Save current session in well-known location and provide context management.
+class SessionContextFactory:
+    def __init__(self) -> None:
+        self.session_cls = None
+        self.expire_on_commit = False
 
-    New in version 3.0.0:
-        Now thread safe using a `scoped_session`.
-    """
-    session = None
```

Comment:
This was still a global before so it did not allow true thread safety

to do so we needed to make session a property